lib/codecs/src/decoding/format/gelf.rs

```diff
@@ -0,0 +1,203 @@
+        }
+
+        if let Some(timestamp) = parsed.timestamp {
+            let naive = NaiveDateTime::from_timestamp(
+                f64::trunc(timestamp) as i64,
+                f64::fract(timestamp) as u32,
+            );
+            log.insert(
+                log_schema().timestamp_key(),
+                DateTime::<Utc>::from_utc(naive, Utc),
+            );
+        } else {
+            log.insert(log_schema().timestamp_key(), Utc::now());
+        }
+
+        if let Some(level) = parsed.level {
+            log.insert(LEVEL, level);
+        }
+        if let Some(facility) = &parsed.facility {
+            log.insert(FACILITY, facility.to_string());
+        }
+        if let Some(line) = &parsed.line {
+            log.insert(LINE, *line);
+        }
+        if let Some(file) = &parsed.file {
+            log.insert(FILE, file.to_string());
+        }
+
+        if let Some(add) = &parsed.additional_fields {
+            for (key, val) in add.iter() {
+                // per GELF spec, filter out _id
+                if key == "_id" {
+                    continue;
+                }
+                // per GELF spec, Additional field names must be characters dashes or dots
+
+                // TODO currently dropping fields that don't match the GELF spec... should we error instead?
+                if !self.regex.is_match(key) {
+                    continue;
+                }
+                // TODO Vector seems to not accept "." or "-" for log field names...
+                // not sure if this is the best approach but seems either we will need to modify the
+                // field names we received, or alter vector to allow "." or "-" in the field names.
+                let key = key.replace('.', "_");
+                let key = key.replace('-', "_");
+
+                // per GELF spec, values to additional fields can be strings or numbers
+                if val.is_string() {
+                    log.insert(key.as_str(), val.as_str());
+                } else if val.is_number() {
+                    log.insert(key.as_str(), val.as_i64());
+                }
+            }
+        }
+
+        Ok(())
+    }
+}
+
+#[derive(Serialize, Deserialize)]
+struct GelfMessage {
+    version: String,
+    host: String,
+    short_message: String,
+    full_message: Option<String>,
+    timestamp: Option<f64>,
+    level: Option<u8>,
+    facility: Option<String>,
+    line: Option<usize>,
+    file: Option<String>,
+    #[serde(flatten)]
+    additional_fields: Option<HashMap<String, serde_json::Value>>,
+}
+
+impl Deserializer for GelfDeserializer {
+    fn parse(&self, bytes: Bytes) -> vector_core::Result<SmallVec<[Event; 1]>> {
+        let line = std::str::from_utf8(&bytes)?;
+        let line = line.trim();
+
+        let parsed: GelfMessage = serde_json::from_str(line)?;
+        let mut event = Event::from(parsed.short_message.to_string());
+        self.insert_fields_from_gelf(&mut event, &parsed)?;
```

Comment:
Nit: I'd probably have the function take `GelfMessage` and return `Event` rather than building the event here and passing it in to be mutated. I think that is easier to follow.

Reply:
Will change it. Was mirroring the conventions in the syslog decoder.